waspc/data/Generator/templates/server/src/auth/providers/oauth/user.ts

```diff
@@ -81,6 +84,8 @@ async function getAuthIdFromProviderDetails(
       userFields as any,
     )
 
+    // postSignup(providerId)
```

Comment:
The first time the OAuth user is encountered, it's created, so the `postSignup` hook is executed.